### ISSUE TITLE
Immediate yield on eager thunk execution grants better user experience on proc count

### ISSUE THREAD

**Issue by krynju** [JuliaParallel/Dagger.jl#228]
I noticed that when creating a lot of eager thunks at once the task scheduler will go and start executing them before the main thread finishes spawning all the eager thunks. That locks the environment for the user and they have to wait until the finish of creation and execution of majority of thunks.

So what I did is in the function wrapped using the Dagger.@spawn macro I put yield() in the beginning, so that the task immediately stops and the thread gets back to creating more eager thunks (and then when creation is done the thread gets busy with the created tasks)

This would be helpful for running on a one thread setup. Either have it as a yield() built into the macro in case the user is running on one thread or maybe have some switch to stop the scheduler from actually running the thunks, so that the creation of them can happen quickly

```julia
# quicker to create these if a lot of them are being created at once
Dagger.@spawn begin 
    yield()
    f() #heavier fun
end
```

I'll get a good example to reproduce this later

**Comment by jpsamaroo** [comment]
I think it really depends on how yield-heavy the task is. If you only have 1 thread and you start a 30-second-long non-yielding operation, at some point you'll have to wait 30 seconds, and it could happen at any time (since there's no way to delay an eager thunk from being scheduled). What we really want is one or more dedicated scheduling threads, which we could accomplish with thread pools. Still, that would require that the user starts more than 1 thread to be useful.

**Comment by krynju** [comment]
Yeah, didn't think about that. That execution thread is shared between the REPL and Dagger scheduler, so the REPL is bound to lock up for the user on single core no matter the approach. 

So you mean is to mark a subset of the threads available as scheduling threads, so they're excluded from getting  execution tasks assigned to? The rest of threads would then become the thread pool for execution. 
Seems that can be done within Dagger, but it would be nice to have it Julia-wide, so that the threads are reserved properly for their use case (like REPL would always have a thread reserved to work on for the interactive session things).

**Comment by jpsamaroo** [comment]
We're basically waiting on an API in Julia that'll let us create our own set of threads that no other tasks can be scheduled on. Until then, I think I'd prefer we leave things as they are.

**Comment by krynju** [comment]
Makes sense. Closing then!
